wd_parallel_pkg/wd_parallel/apply.py

```
@@ -0,0 +1,347 @@
+        elif partial:
+            dense_sum.append((p, grad))
+        else:
+            dense_mean.append((p, grad))
+
+    # Coalesced all-reduce for mean params (the common case, one collective).
+    if dense_mean:
+        flat = torch.cat([g.contiguous().view(-1) for _, g in dense_mean])
+        dist.all_reduce(flat, op=dist.ReduceOp.SUM, group=group)
+        flat.div_(world_size)
+        offset = 0
+        for p, g in dense_mean:
```

Ruff (0.15.11)

[warning] Loop control variable `p` not used within loop body

(B007)